### PR TITLE
add a way to disable PTY allocation

### LINKED ISSUE #179: add a way to disable PTY allocation
One of the commands Ansible invokes on target system is `/bin/sh -c '/usr/bin/python && sleep 0'` which is used to determine whether `python` executable is present on the target. Unfortunately, if the pty is allocated, the REPL is invoked which causes the process to fail.

Would it be possible to add an option to disable PTY allocation? I am okay creating the patch myself: unfortunately I haven't figured out the way how it's possible to change the runtime spec.

### PR DESCRIPTION
**Pull request by TomasTomecek**
Fixes #179

This patch resolves the original issue: https://github.com/ansible/ansible/pull/26170#issuecomment-312591504

Signed-off-by: Tomas Tomecek <ttomecek@redhat.com>

### CODE REVIEW

File: docs/buildah-run.md [hunk 1]
```diff
@@ -14,6 +14,12 @@ the *buildah config* command.
 
 ## OPTIONS
 
+**--tty**
```
Comment (rhatdan) [inline docs/buildah-run.md]: How about 
```
**--tty**

By default a pseudo-TTY is allocated only when buildah's standard output is
attached to a pseudo-TTY.  Setting the `--tty` option to `true` will cause a
pseudo-TTY to be allocated inside the container. Setting the `--tty` option to `false` will prevent the pseudo-TTY from being allocated.
```

**Comment by rhatdan** [comment]
Do we need the same feature for buildah bud?


We also need bash completion.

File: docs/buildah-run.md (hunk 1 above)
Comment (nalind) [inline docs/buildah-run.md]: ... standard output is attached to a terminal.  Though we might want to change it to point to check standard input instead.

Comment (TomasTomecek) [inline docs/buildah-run.md]: I like your suggestions.

**Comment by TomasTomecek** [comment]
> Do we need the same feature for buildah bud?

`docker build` doesn't have such option. I suspect it may not be needed. The only problem may be if some tools try to be interactive (that's what I run into with `run`). Not sure if PTY is allocated by default.

I think I addressed all the comments.

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@nalind Do you want to merge?

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by nalind** [comment]
@rhatdan Yup, looks good to me.  Merging.

**Comment by nalind** [comment]
@rh-atomic-bot r+ 62b79edc26c95ac85dfe898af91ad8145ed90377

**Pull request closed without merging**